bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -56,37 +59,96 @@ export class StringFormatter extends CellFormatter {
   }
 
   static {
-    this.define<StringFormatter.Props>(({Color, Nullable, String}) => ({
-      font_style: [ FontStyle, "normal" ],
-      text_align: [ TextAlign, "left"   ],
-      text_color: [ Nullable(Color), null ],
+    this.define<StringFormatter.Props>(({String}) => ({
+      font_style: [ p.FontStyleSpec, {value: "normal"} ],
+      text_align: [ p.TextAlignSpec, {value: "left"} ],
+      text_color: [ p.ColorSpec, null ],
+      background_color: [ p.ColorSpec, null ],
       nan_format: [ String, "-"],
     }))
   }
 
   override doFormat(_row: any, _cell: any, value: any, _columnDef: any, _dataContext: any): string {
-    const {font_style, text_align, text_color} = this
+    const {font_style, text_align, text_color, background_color} = this
 
     const text = div(value == null ? "" : `${value}`)
-    switch (font_style) {
+
+    // Font style
+    let resolved_font_style
+    if (isValue(font_style)) {
+      resolved_font_style = font_style.value
+    } else if (isField(font_style)) {
+      resolved_font_style = _dataContext[font_style.field]
+    } else if (isExpr(font_style)) {
+      // TODO
+    } else {
+      unreachable()
+    }
+
+    switch (resolved_font_style) {
       case "normal":
-        break
-      case "bold":
-        text.style.fontWeight = "bold"
+        // nothing to do
         break
       case "italic":
         text.style.fontStyle = "italic"
         break
-      case "bold italic":
+      case "bold":
         text.style.fontWeight = "bold"
+        break
+      case "bold italic":
         text.style.fontStyle = "italic"
+        text.style.fontWeight = "bold"
         break
     }
 
-    text.style.textAlign = text_align
+    // Text align
+    if (isValue(text_align)) {
+      text.style.textAlign = text_align.value
+    } else if (isField(text_align)) {
+      text.style.textAlign = _dataContext[text_align.field]
+    } else if (isExpr(text_align)) {
+      // TODO
+    } else {
+      unreachable()
+    }
+
+    // Text color
+    // Handle the most common case first : isValue and value == null
+    if (isValue(text_color)) {
+      if (text_color.value != null) {
+        text.style.color = color2css(text_color.value)
+      }
+    } else if (isField(text_color)) {
+      if (text_color.transform != null && text_color.transform instanceof ColorMapper) {
+        const rgbaArray = text_color.transform.rgba_mapper.v_compute([_dataContext[text_color.field]])
+        const [r, g, b, a] = rgbaArray
+        text.style.color = rgba2css([r, g, b, a])
+      } else {
+        text.style.color = _dataContext[text_color.field]
```

Comment:
A data point in a color array can be almost anything, so this should also go through `color2css()` as well.

Reply:
Same applies to `background_color`.